### DISCUSSION TITLE
An error occurred when cross compiling gcc

### DISCUSSION THREAD

**Discussion by peakedshout** [fyne-io/fyne#4840] [Q&A]
I tried to compile programs on darwin, linux, and windows platforms on Linux, but gcc reported an error.
```
darwin:
fyne package -os darwin -icon ./resource/home.png
# runtime/cgo
gcc: error: unrecognized command-line option '-arch'
gcc: error: unrecognized command-line option '-mmacosx-version-min=10.11'
windows:
fyne package -os windows -icon ./resource/home.png
# runtime/cgo
gcc: error: unrecognized command-line option ‘-mthreads’; did you mean ‘-pthread’?
```

In the end, only linux succeeded, of course because I am doing compilation using linux.
My gcc version is 4:11.2.0-1ubuntu1  (I tried updating gcc but it prompts me that this is the latest)
I don't quite understand how to deal with this problem? My program is running normally.

**Comment by andydotxyz** [comment]
To build for darwin needs a different C compiler - you will need to install that and set CC as described in https://docs.fyne.io/started/cross-compiling

Or use fyne-cross, linked at the end of the article.

**Reply by peakedshout** [reply]
Thank! I'll check it out, but I still have a question about memory. I wrote a simple program, which is not complicated in my opinion, but uses a lot of memory, on linux, there are more than 100 MB with casual use. And if I set the font via environment variables, then the memory increases to more than 260! Is this normal?

**Reply by peakedshout** [reply]
This is still the memory dedicated to the pure computing program, and the shared memory remains unchanged at 90 MB. I'm worried about this increase in dedicated memory.

**Reply by dweymouth** [reply]
About memory use, there are things we are tracking for future improvements - text caching will be one of them which will be made more efficient in the future. But it is possible your app can contribute to high memory use as well by doing things like, for example, creating new widgets rather than just updating their state when you want to display new content

**Reply by dweymouth** [reply]
If you're not on the latest Fyne v2.4.5, it's also possible the high memory is caused by #4723 which was recently fixed

**Reply by peakedshout** [reply]
I don't think my own program will have memory leaks, because I basically use simple HTTP. However, instead of building new ones, I have the behavior of refreshing components on a regular basis, and yes, I've been reusing components, and when the program starts running, I build all the components I need and don't build along the way. On top of that, I'm more curious about why loading external fonts results in such a large memory usage, the font itself is only 10MB, and the version I'm using is also the latest.

**Reply by dweymouth** [reply]
The font has to be uncompressed and loaded into data structures in memory, so it will use more RAM than the size of the font file. It's possible there can be some further improvements in the future though. And starting in the upcoming Fyne 2.5, it will look up fonts from the OS as needed so you may no longer need to bundle a custom font with your app.

**Reply by peakedshout** [reply]
Haha, that's so much to look forward to!
As an aside, I found a little problem on the documentation on the official website. I'm using fyne-cross for cross-compilation, but it shows up in darwin
```
[✗] could not make command context for darwin OS: arch "386" is not supported. Supported: [amd64 arm64]
```
But the official website says so, I think it should be a problem that it has not been updated?
```
GOOS | GOARCH
-- | --
darwin | amd64
darwin | 386

```

**Comment by peakedshout** [comment]
Fyne-Cross's Android image doesn't support the ARM version, and I can't run the image on macOS, which frustrates me.

**Reply by andydotxyz** [reply]
Can you be more specific? the arm version of what? macOS won't work for what reason?
I'm not sure we can action these frustrations without more info.

**Reply by peakedshout** [reply]
```
fyne-cross android -arch=amd64,386,arm64,arm
```
I'm on my macOS M3 I tried to run this but fyneio/fyne-cross-images:android only amd64 and I can't run it on my arm64 machine

**Reply by andydotxyz** [reply]
Ah I understand. I think that is captured by an issue in fyne-cross project https://github.com/fyne-io/fyne-cross/issues/219.
